CRSF sensor page added

jlpoltrack opened the pull request:
New page with CRSF sensors.

Notes:
- There are 2 'TPWR' datapoints, verified on two transmitters.
- Not sure on 'ANT' datapoint

olliw42 commented:
good idea!

concerning the 2 TWPR, this is a EdgeTx bug, in OpenTx they show up as TWPR and TPw2 (which implies that you need to adapt the docs somehow to reflect this)

ANT is not supported (= always 0)(I think we once discussed that it should show the receiver's receive antenna, which is however not available)(maybe we should change the protocol a bit to yield more of this info)

there a couple more fields which are nopt supported, you can infer from here https://github.com/olliw42/mLRS/blob/main/mLRS/CommonTx/crsf_interface_tx.h#L784-L862

some fields may show a value which is not accurate however, e.g. TFPS shows 20,30,50 instead of 19,31,50. Similary, RFMD shows the CRSF value for the 50 Hz mode, but artifical 19,31 for the 19/31Hz modes. There might some few more fields there one has to do such adaption trickery

jlpoltrack commented:
Thanks for the feedback, made the following updates:

- Added note on TPWR / TPw2
- Updated sensors that aren't supported (2RSS, ANT, RSNR) per the link
- Expanded description for RFMD, TFPS
- Small tweak to language, from 'supports 28 sensors' to 'provides 28 sensors'

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,40 @@
+1   | 1RSS       | Receiver RSSI                                                           | mLRS        |
+2   | 2RSS       | Not supported                                                           |             |
+3   | Alt        | Altitude                                                                | FC          |
+4   | ANT        | Not supported                                                           |             |
```
I just pushed a commit to support ANT, it is the receiver seide receive antenna now

jlpoltrack replied:
Nice, added this.

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,40 @@
+10  | GSpd       | Ground speed                                                            | FC          |
+11  | Hdg        | Heading                                                                 | FC          |
+12  | Ptch       | FC pitch angle                                                          | FC          |
+13  | RFMD       | Receiver update rate (19, 31, 50)                                       | mLRS        |
```
the values actually are 19, 31, 1 !!! This is because in CRSF language value 1 is for 50 Hz ... !!!
I'm not sure from top of my head if OTX/ETX display 1 as 50 Hz, pl check what is actually displayed

jlpoltrack replied:
You're right - 50Hz shows up as 1.  Added this to the description.

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,40 @@
+18  | RxBt       | Battery Voltage                                                         | FC          |
+19  | Sats       | GPS satellites acquired                                                 | FC          |
+20  | TFPS       | Transmitter update rate (20, 30, 50 - use RFMD instead)                 | mLRS        |
+21  | TPWR       | Transmitter power                                                       | mLRS        |
```
maybe indicate the unit of the power

jlpoltrack replied:
Added mW.

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,40 @@
+19  | Sats       | GPS satellites acquired                                                 | FC          |
+20  | TFPS       | Transmitter update rate (20, 30, 50 - use RFMD instead)                 | mLRS        |
+21  | TPWR       | Transmitter power                                                       | mLRS        |
+22  | TPw2*      | Transmitter power                                                       | mLRS        |
```
dito

jlpoltrack replied:
Added.  FYI, CRSF power levels are out of order :) {0,10,25,100,500,1000, 2000,250,50}

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,40 @@
+
+Num | Datapoint | Description                                                              | Data Source |
+--- | --------- | ------------------------------------------------------------------------ | ----------- |
+1   | 1RSS       | Receiver RSSI                                                           | mLRS        |
```
I think we want to mention that it is the negative of the RSSI! (to avoid possible confusion with other sources of RSSI)
also applies to RSSI's below
maybe a footnote?

jlpoltrack replied:
Added note: 'RSSI values are displayed as positive values, e.g. -85dBm is displayed as 85dBm'

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,41 @@
+24  | TRSP        | Transmitter RSSI percentage                                             | mLRS        |
+25  | TRSS*       | Transmitter RSSI                                                        | mLRS        |
+26  | TSNR        | Transmitter signal to noise ratio                                       | mLRS        |
+27  | VSpd        | Vertical speed (barometer)                                              | FC          |
```
this is actually not the barometer (there is an extra BARO_ALT crsf message for that specifically)
it's what ardupilot comes up with after all its sensor fusion EKF thingy

jlpoltrack replied:
Will remove the the '(barometer' piece

olliw42 commented on docs/CRSF_SENSORS.md:
```diff
@@ -0,0 +1,41 @@
+10  | GSpd        | Ground speed                                                            | FC          |
+11  | Hdg         | Heading                                                                 | FC          |
+12  | Ptch        | FC pitch angle                                                          | FC          |
+13  | RFMD        | Receiver update rate (19, 31, 1 - 50Hz)                                 | mLRS        |
```
we could change the code so that 50 indeed shows up as 50 ... might be incompatible with some other lua stuff, I don't know
what would you say?

jlpoltrack replied:
Do you want to check to see if that change would break Yaapu?  I took a quick look at the code and he looks for RFMD == 1 for one of these 2Hz tasks.  https://github.com/yaapu/FrskyTelemetryScript/blob/master/OTX_ETX/c480x272/SD/WIDGETS/yaapu/main.lua#L2255

Maybe just leave alone?

olliw42 replied:
ah, ok, so lets leave it as it is :)

olliw42 commented:
looks great
many thx
shall I merge?

jlpoltrack commented:
Good to merge.

olliw42 commented:
MANY THX sir :)
